core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -386,7 +387,7 @@ public final void runInContext() {
             }
           } catch (Throwable t) {
             stream.cancel(Status.CANCELLED.withCause(t).withDescription("Failed to read message."));
-            return;
+            Throwables.propagateIfPossible(t);
```

Comment:
The goal is to get the exception logged.

Did I miss somewhere that the `stream.cancel` call will cause something to be logged somewhere in the running process? I don't see it but maybe I missed something.

AFAICT, as it stands right now, an NPE in an `onNext` callback is sent on the stream but never noted locally. In my case, it's the RPC result delivery `onNext` so the server side doesn't see it? It certainly doesn't care if the client throws and exception, or at least why.

Defeating logging of `Error`s, let alone `RuntimeException`s, seems like a bad thing?

With the patch, a `RuntimeException` is caught and logged at
https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/SerializingExecutor.java#L157

`Error`s aren't caught and so propagate to the default JVM uncaught exception handler which logs them.

Both of these behaviors are good by me.

Not logging an exception like an NPE in the process that throws it seems like a Bad Thing ... and cost me a fair amount of time today. A result wasn't being processed because of an exception in an `onNext` handler but there was no indication anywhere that this was happening. It looked like the response was never being sent so I spent a fair amount of time figuring out where the dataflow break was occurring.